gson/src/main/java/com/google/gson/JsonObject.java

```diff
@@ -153,13 +155,40 @@ public JsonElement get(String memberName) {
   }
 
   /**
+   * Returns the member with the specified name if it exists, otherwise the supplied default value
+   * is returned.
+   *
+   * @param memberName name of the member that is being requested.
+   * @param defaultValue the value to be returned if the member does not exist.
+   * @return the member matching the name. <code>defaultValue</code> if no such member exists.
+   */
+  public JsonElement get(String memberName, JsonElement defaultValue) {
+    JsonElement result = get(memberName);
+    return result != null ? result : defaultValue;
+  }
+
+  /**
    * Convenience method to get the specified member as a JsonPrimitive element.
    *
    * @param memberName name of the member being requested.
    * @return the JsonPrimitive corresponding to the specified member.
    */
   public JsonPrimitive getAsJsonPrimitive(String memberName) {
-    return (JsonPrimitive) members.get(memberName);
+    return (JsonPrimitive) get(memberName);
+  }
+
+  /**
+   * Convenience method to get the specified member as a JsonPrimitive element if it exists,
+   * otherwise the supplied default value is returned.
+   *
+   * @param memberName name of the member being requested.
+   * @param defaultValue the value to be returned if the member does not exist.
+   * @return the JsonPrimitive corresponding to the specified member. <code>defaultValue</code>
+   * if no such member exists.
+   */
+  public JsonPrimitive getAsJsonPrimitive(String memberName, JsonPrimitive defaultValue) {
```

Comment:
Not too sure about the advantages of default value methods. Too many new methods get added for a small benefit.